### ISSUE TITLE
Remove dependence on Bootstrap

### ISSUE THREAD

**Issue by benwerd** [idno/known#2617]
No official Known template should use Bootstrap on any level. Instead, we should use CSS grids and a simple, sensible base CSS stylesheet.

**Comment by mapkyca** [comment]
I'd also like to remove the dependence on jquery

**Comment by benwerd** [comment]
@mapkyca +1 - adding that to the roadmap too

**Comment by benwerd** [comment]
See #2622

**Comment by johanbove** [comment]
Even-though I agree with removing Bootstrap and jQuery dependencies, I would still advocate in keeping a CSS system, so all themes and core templates speak the same "CSS language". Would you consider TailWindCSS? https://tailwindcss.com/

**Comment by mapkyca** [comment]
We'll use sass, and whatever framework makes sense to give us a leg up. 

I have a strong view that we actually make the front end skin a separate repo, that way the dependency management between core and plugins can be offloaded to composer.

Themes would build against a specific front end skin, meaning we can run multiple engines concurrently.

**Comment by mapkyca** [comment]
I will also add that, where practical, we should move templating over to twig at this time. It's mature, widely used, more secure, and cross platform (so we could use the same template tools for e.g. react)

**Comment by benwerd** [comment]
I’ll do some reading on Twig and React. It wasn’t in a place where expressive development was possible with it when I started work on Known, but I’d love it if it had evolved to work well with modern development practices. The thing I’m looking for is cascading templates: the ability to supersede just the templates you need to in order to create a new theme. Otherwise we have a ton of duplicated code in each new theme, which is impractical and messy (and potentially, in the worst case, a security issue).

**Comment by mapkyca** [comment]
Yep yep. Totally possible right now with the hybrid template class right now in known (if there's a twig template, it'll be used, just like a standard bonita template). 

It's a fairly simplistic approach, and there's lots that twig can do that we might not be able to use with this approach (like twig native inserts and overrides), but as I look into it more we can do more. 

My point is that we should be consistent with templating and not reinvent the wheel - twig is mature and supported both in javascript and for classic back end generated dom.

**Comment by physcocode** [comment]
+1 for twig I am up for that
